### ISSUE TITLE
Add support for `IP_RECVERR` and `IPV6_RECVERR`

### ISSUE THREAD

**Issue by larseggert** [quinn-rs/quinn#2052]
Would be useful, for example, to receive `fragmentation needed and DF set` ICMP messages.

**Comment by mxinden** [comment]
See draft implementation by @aaron-ang in https://github.com/quinn-rs/quinn/pull/2133.

**Comment by AnkurRathore** [comment]
I would like to pickup this issue. Let me know what process to follow to start contributing. I am familiar with Rust, looking to start contributing on some good open source projects.

**Comment by aaron-ang** [comment]
> I would like to pickup this issue. Let me know what process to follow to start contributing. I am familiar with Rust, looking to start contributing on some good open source projects.

Please go ahead.

**Comment by AnkurRathore** [comment]
@mxinden and @aaron-ang is this issue still open?

**Comment by mxinden** [comment]
Yes, that is my understanding @AnkurRathore. Please go ahead and submit a pull request if you want.

**Comment by RaghavendraRQ** [comment]
I would like to inquire if the issue is still **open**. If so, I would be interested in taking it on. Thank you for your consideration.

**Comment by mxinden** [comment]
@RaghavendraRQ yes, still open and I think worth exploring.

**Comment by syszery** [comment]
I’m currently working on this as discussed in #2421.
